### PR TITLE
Update emacs and gemacs to 26.2 and use gtk+ v3 instead of lucid

### LINKED ISSUE #3253: emacs and gemacs 26.2
I previously had troubles with menu-bar fonts using emacs and gemacs 26.1 compiled with lucid. I managed to have a full running emacs and gemacs using 26.2 and gtk+ v3 which allows to display images and manage fonts in a functional way.

Here is what I have done : 
```ruby
require 'package'

class Gemacs < Package
  description 'An extensible, customizable, free/libre text editor - and more.'
  homepage 'http://www.gnu.org/software/emacs/'
  version '26.2'
  source_url 'https://ftp.gnu.org/gnu/emacs/emacs-26.2.tar.xz'
  source_sha256 '151ce69dbe5b809d4492ffae4a4b153b2778459de6deb26f35691e1281a9c58e'
  #source_sha256 '1cf4fc240cd77c25309d15e18593789c8dbfba5c2b44d8f77c886542300fd32c'

  binary_url ({
    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/gemacs-26.1-chromeos-armv7l.tar.xz',
     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/gemacs-26.1-chromeos-armv7l.tar.xz',
       i686: 'https://dl.bintray.com/chromebrew/chromebrew/gemacs-26.1-chromeos-i686.tar.xz',
     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/gemacs-26.1-chromeos-x86_64.tar.xz',
  })
  binary_sha256 ({
    aarch64: '4ded760607b64f706cc83b92042609845f8213a697bc82a931f04ddcf1c94a17',
     armv7l: '4ded760607b64f706cc83b92042609845f8213a697bc82a931f04ddcf1c94a17',
       i686: 'f8c514abd49a4a8aefa6235918d442d75d1a722e9558cd741246769cfeb9c8da',
     x86_64: '7d646d49bec0acf2c7eb79ff589c6f311537233565f43646dafa927de1adbb8c',
  })

  depends_on 'emacs'
  depends_on 'giflib'
  depends_on 'libjpeg'
  depends_on 'libpng'
  depends_on 'libtiff'
  depends_on 'imagemagick6'
  depends_on 'sommelier'

  def self.build
    system "./configure \
            --prefix=#{CREW_PREFIX} \
            --localstatedir=#{CREW_PREFIX}/share \
            --with-x=yes \
            --with-x-toolkit=gtk3 \
            --with-gif=yes \
	    --with-jpeg=yes \
	    --with-png=yes \
	    --with-rsvg=yes" 
    system 'make'
  end

  def self.install
    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
    system "rm -rf #{CREW_DEST_DIR}"
    system "install -Dm755 src/emacs #{CREW_DEST_PREFIX}/bin/gemacs"
    system "install -Dm755 src/emacs-26.2.1 #{CREW_DEST_PREFIX}/bin/gemacs-26.2.1"
  end
end
```
and
```ruby
require 'package'

class Emacs < Package
  description 'An extensible, customizable, free/libre text editor - and more.'
  homepage 'http://www.gnu.org/software/emacs/'
  version '26.2'
  source_url 'https://ftp.gnu.org/gnu/emacs/emacs-26.2.tar.xz'
  source_sha256 '151ce69dbe5b809d4492ffae4a4b153b2778459de6deb26f35691e1281a9c58e'
  #source_sha256 '1cf4fc240cd77c25309d15e18593789c8dbfba5c2b44d8f77c886542300fd32c'

  binary_url ({
    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/emacs-26.1-1-chromeos-armv7l.tar.xz',
     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/emacs-26.1-1-chromeos-armv7l.tar.xz',
       i686: 'https://dl.bintray.com/chromebrew/chromebrew/emacs-26.1-1-chromeos-i686.tar.xz',
     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/emacs-26.1-1-chromeos-x86_64.tar.xz',
  })
  binary_sha256 ({
    aarch64: 'f932a0b4ef4e33e230abd07d39ba12e6661304b2baaddacc5f03470d355dd021',
     armv7l: 'f932a0b4ef4e33e230abd07d39ba12e6661304b2baaddacc5f03470d355dd021',
       i686: 'd5b9fed4b9f42e04da7b326d339e23a1b6cf0367885a413e5dfbff4f90c9feef',
     x86_64: '20adfa13c8b999da71d4efe1c6e86521ad83481e1d593bdcab20eac5c2cb589e',
  })

  depends_on 'lcms'
  depends_on 'libjpeg'

  def self.build
    system "./configure \
            --prefix=#{CREW_PREFIX} \
            --localstatedir=#{CREW_PREFIX}/share \
            --with-x=no \
            --without-makeinfo \
            --without-selinux"
    system 'make'
  end

  def self.install
    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
  end
end
```
then I remove any emacs or gemacs installation and ran 
```bash
crew install -s emacs gemacs
```

Is it worth it to upgrade emacs and gemacs packages and migrate from lucid to gtk+v3 ?

### PR DESCRIPTION
**Pull request by vincowl**
Fixes #3253 

I previously had troubles with menu-bar fonts using emacs and gemacs 26.1 compiled with lucid. I managed to have a full running emacs and gemacs using 26.2 and gtk+ v3 which allows to display images and manage fonts in a functional way.

### CODE REVIEW

File: packages/gemacs.rb [hunk 1]
```diff
@@ -3,45 +3,42 @@
...
-     armv7l: '4ded760607b64f706cc83b92042609845f8213a697bc82a931f04ddcf1c94a17',
-       i686: 'f8c514abd49a4a8aefa6235918d442d75d1a722e9558cd741246769cfeb9c8da',
-     x86_64: '7d646d49bec0acf2c7eb79ff589c6f311537233565f43646dafa927de1adbb8c',
+
   })
 
   depends_on 'emacs'
   depends_on 'giflib'
   depends_on 'libjpeg'
+  depends_on 'libpng'
```
Comment (JL2210) [inline packages/gemacs.rb]: This is redundant since you include the dependency in the `emacs` package.

**Comment by JL2210** [comment]
Please address above comments and then this will be good to merge.

**Comment by vincowl** [comment]
I let gemacs dependencies as it was, but I don't really understand why it depends on emacs package since we are rebuilding the same sources. Any idea ? I think gemacs and emacs are exactly the same sources which are built with different options and thus different dependancies for graphical components (imagemagick, ...)

**Comment by JL2210** [comment]
People expect the `emacs` command and manpages with `gemacs`,

File: packages/gemacs.rb [hunk 2]
```diff
@@ -3,45 +3,42 @@
...
-            --without-selinux"
+	    --with-jpeg=yes \
+	    --with-png=yes \
+	    --with-rsvg=yes" 
     system 'make'
   end
 
   def self.install
     system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
     system "rm -rf #{CREW_DEST_DIR}"
```
Comment (uberhacker) [inline packages/gemacs.rb]: The 2 lines above can be removed.  It doesn't make sense to install and remove immediately afterwards.

**Comment by uberhacker** [comment]
@vincowl: The gemacs package only installs 3 additional files that complement the emacs package.  The emacs package acts as "core" for gemacs.  Hopefully, that makes sense.

File: packages/gemacs.rb (hunk 1 above)
Comment (uberhacker) [reply packages/gemacs.rb]: Actually, `depends_on 'libjpeg'` and `depends_on 'libpng'` can both be excluded.

**Comment by vincowl** [comment]
To answer to the previous remarks, I made a new commit since I tried something different. I think we don't need to depend on `emacs` package and make `emacs` and `gemacs` two different products. That's why I made this new commit to fullfill previous requirements : also get `emacs` command and mapages.  This is fully functional. But maybe `gemacs` was designed to be like an upper layer to `emacs`. But this would  need to maintain these two packages in a very tight way.

**Comment by JL2210** [comment]
Not a good idea. We tried that with vim and it breaks stuff if people
install both packages and remove one of them. Just keep it the way it is.

**Comment by vincowl** [comment]
@JL2210 All right, makes sense. I'll keep it as it was with above requests.

File: packages/gemacs.rb (hunk 2 above)
Comment (vincowl) [reply packages/gemacs.rb]: Taken into account in 0ffa90c

File: packages/gemacs.rb (hunk 1 above)
Comment (vincowl) [reply packages/gemacs.rb]: Taken into account in 0ffa90c

**Comment by vincowl** [comment]
Should normally be good to merge with 0ffa90c

**Comment by vincowl** [comment]
@JL2210 `lcms` is here to use Little CMS a color management engine

**Comment by vincowl** [comment]
Just a simple question : how to we have to proceed to create binaries in order to add them to the `binary url` part ?

**Comment by JL2210** [comment]
Don't bother. Unless you can get your hands on all 3 of the Chromebook
architectures, it's just a lot easier to let someone else do it.

**Comment by vincowl** [comment]
😄 That suits me perfectly well ! The less work for me the better !!

**Pull request merged**